data/finset.lean

```diff
@@ -1420,3 +1420,37 @@ begin
 end
 
 end list
+
+namespace finset
+
+/-- A type is (constructively) infinite if it can be shown that, for any finite
+set of inhabitants of that type, there exists an inhabitant not in the set. -/
+def infinite (α : Type*) : Prop :=
+∀ (s : finset α), ∃ (a : α), a ∉ s
```

Comment:
`infinite` looks like it should be a type class (corresponding to `fintype`).

Maybe also make it into `type`, as it is a constructive version?

Reply:
> `infinite` looks like it should be a type class (corresponding to `fintype`).

I wasn't sure about that. There are certainly multiple ways to prove that a type is infinite, which means you can have multiple inhabitants of `infinite α` for a given `α`. I don't know how significant that is. If it's not important, then a type class might make sense.

> Maybe also make it into `type`, as it is a constructive version?

Do you mean name it `infinitetype`, or do you mean use `Type` instead of `Prop`?

Reply:
Keep the name. But instead of being the `finset` namespace, but it in the toplevel and use `class` instead of `def`.

I'm not sure myself if it should be `Prop` or `Type`. For first lets stay with `Prop`.